### PR TITLE
Use Moment struct in WuQuantizer.

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Very simple PR. Refactors the `WuFrameQuantizer<TPixel>` to us a single pooled buffer instead of many. Makes code much easier to follow, improves performance, and reduces GC pressure.

### CODE REVIEW

File: src/ImageSharp/Processing/Processors/Quantization/WuFrameQuantizer{TPixel}.cs [hunk 1]
```diff
@@ -199,27 +155,18 @@ protected override ReadOnlyMemory<TPixel> GetPalette()
...
 
                 for (int k = 0; k < this.colors; k++)
                 {
                     this.Mark(ref this.colorCube[k], (byte)k);
 
-                    float weight = Volume(ref this.colorCube[k], vwtSpan);
+                    Moment moment = Volume(ref this.colorCube[k], momentsSpan);
 
-                    if (MathF.Abs(weight) > Constants.Epsilon)
+                    if (moment.Weight > 0)
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Quantization/WuFrameQuantizer{TPixel}.cs]: I'm not familiar with the algorithm, but removing `Epsilon` changes the algorithm a bit. Are there any corner cases where this may have an impact?

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Quantization/WuFrameQuantizer{TPixel}.cs]: None that I could find or test. I know the algorithm pretty well now and the epsilons were actually added as part of some bad design from me in the past. Weight should be a `long`.

Comment (antonfirsov) [reply src/ImageSharp/Processing/Processors/Quantization/WuFrameQuantizer{TPixel}.cs]: Cool! If the struct is uniform (all members Int64), it opens up nice SIMD optimization opportunities.
